Lib/fileinput.py

```diff
@@ -348,14 +349,14 @@ def _readline(self):
                 try:
                     perm = os.fstat(self._file.fileno()).st_mode
                 except OSError:
-                    self._output = open(self._filename, "w")
+                    self._output = open(self._filename, self._write_mode)
                 else:
                     mode = os.O_CREAT | os.O_WRONLY | os.O_TRUNC
-                    if hasattr(os, 'O_BINARY'):
+                    if hasattr(os, 'O_BINARY') and 'b' in self._mode:
```

Comment:
You need to always add the O_BINARY flag on Windows, otherwise the system will translate newline characters. This does not have relations to binary/text files in Python.

Reply:
Good point, fixed, thank you!